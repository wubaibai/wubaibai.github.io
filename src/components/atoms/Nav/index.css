.nav {
	font-family: 'Montserrat', sans-serif;
	font-size: 16px;
	font-weight: 400;
	line-height: 1.2;
	text-align: left;
	color: #f2eeec;
}
