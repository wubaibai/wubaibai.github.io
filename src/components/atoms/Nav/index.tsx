import React, { MouseEvent } from 'react';

import { useHistory } from 'models/routing';
import { History } from 'history';

import style from './index.css';

const onClickHandler = (callback: (e: MouseEvent<HTMLAnchorElement>) => void, history: History) => {
	return (e: MouseEvent<HTMLAnchorElement>): void => {
		e.preventDefault();
		history.push(e.currentTarget.pathname);
		callback(e);
	};
};

interface LinkProperty {
	to?: string;
	onClick?: (e: MouseEvent<HTMLAnchorElement>) => void;
	children: React.ReactChildren | string;
}

const Link: React.FC<LinkProperty> = ({ to, onClick = () => {}, children }) => {
	const history = useHistory();

	return (
		<a
			className={style.nav}
			href={to}
			role="button"
			tabIndex={0}
			onClick={onClickHandler(onClick, history)}
			onKeyPress={() => {}}
		>
			{children}
		</a>
	);
};

export default Link;
