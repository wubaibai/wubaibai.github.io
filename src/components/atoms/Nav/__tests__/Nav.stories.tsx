import React from 'react';
import { storiesOf } from '@storybook/react';

import Nav from 'components/atoms/Nav';

const stories = storiesOf('atoms/Nav', module);

stories.add('__interactive', () => <Nav>ABOUT</Nav>, { jest: 'Nav' });
