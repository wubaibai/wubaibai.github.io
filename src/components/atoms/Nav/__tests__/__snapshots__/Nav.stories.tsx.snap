// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <a
    class="nav"
    role="button"
    tabindex="0"
  >
    ABOUT
  </a>
</DocumentFragment>
`;
